### DISCUSSION TITLE
[Bug]: unknown cause codelite or Linux File system corruption

### DISCUSSION THREAD

**Discussion by DavidRamer** [eranif/codelite#3527] [General]
### What happened?

```
ASSERT INFO:
./src/common/fileconf.cpp(1380): assert "!m_pParent || !m_pLine || !pLine" failed in SetLine(): changing line for a non-root group?

BACKTRACE:
[1] wxFileConfig::Parse(wxTextBuffer const&, bool)
[2] wxFileConfig::Init()
[3] wxFileConfig::wxFileConfig(wxString const&, wxString const&, wxString const&, wxString const&, long, wxMBConv const&)
[4] wxAppTraitsBase::CreateConfig()
[5] wxConfigBase::Create()
[6] wxPersistenceManager::RestoreValue(wxPersistentObject const&, wxString const&, long*)
[7] wxEntry(int&, wchar_t**)
[8] __libc_start_main
```

### Version

17.0.0

### Operating system

Linux

### Steps to reproduce

```text
Application compiled without errors. The problem reproduces every time attempt to run program. It occurs while initiating MainFrame GUI. Tried to use debugger to isolate problem but does not get into user written code before error occurs.

I suspected a file system corruption but had difficulty running fsck which always aborted.

Now rather frustrated with situation. Have attempted uninstall/reinstall of codelite but could not get past a lock condition.

Due to my lack of familiarity with Linux getting more info difficult and info on internet either dated or misleading as most people trying to be helpful assume knowledge of Linux environment and therefore very short on actually how to do what they propose as a solution. Therefore if you wish further information such as a dump or log info, please be specific as to where and how to get and provide that info.

Thank you
```

**Comment by eranif** [comment]
Some information is missing: is this coming from CodeLite's code or from your code?
You should try and run it under debugger (e.g. gdb) and get the full backtrace when it asserts (and probably debug with debug info to get source file name + line number)

**Comment by DavidRamer** [comment]
Hello Eran

Thank you for the quick response. I did not expect such a quick response, and I am grateful for your help.

I am using gdb. so the report was produced by gbd. Not much to go by.

I still such a potential corruption of a file or file system, but getting fsck to cooperate is difficult.

One of my peeves this day is the poor quality of error reports from most software. I am an old timer. Been involved in IT for over 50 years. It seems to me that 40 and 50 years ago we had much better error reporting than we see today. For example, we wrote compilers to be efficient and provide accurate information as to why and error occurred and even provided the means to correct the error to finish the compilation and run the program. It was successful, particularly in the university environment where students needed accurate information fast and had little time for investigating why their program had an error.

In contrast, the past few weeks I have been playing with c++ and wxWidgets on Codelite using Gnu cpp. I am not impressed to say the least. A simple error seems to introduce a plethora of other errors, but magically disappear when the real eror is corrected.

I was never a fan of cpp. I had worked with a lot of languages and my experimenting with recently just confirms what I have heard over many years, that it is a difficult language to master.

My academic background is in Comp Sc with a BSc and MSc. I did my thesis under Dr John Peck who was one of the authors of the ALGOL 68 Report and its subsequent Algol 74 Report. ALGOL 68 was a well-designed algorithmic language and is still used as a teaching tool in universities today. I implemented the lexical and syntactic analysis portions of the language and code generation, and fellow student did the semantic analysis. So my background demonstrates why I am so particular about issues of today around language design and the real world of using computer technology.

One area that really turns me off is AI. In 1972, one of my fellow students was a Phd candidate whose goal was to make in difficult for a person to know if they are dealing with a computer or a person. Ironically, that is still AI's goal. Having been exposed to some of that AI technology in a number of different products, I am not impressed as I had to reject at least half or more of its suggests. Personally, I see AI not as a valuable tool in the industry, n=but rather as a political tool to manipulate society. All the hype over AI in the past 5+ years is to attract investor money, but the results will be disasterous. Very little good will come of it. And no I would never trust the code it generates.

Microsoft's CEO recently stated they would fix the security issues with Windows. Well, good luck with that. They would have to redesign the entire operating system from the ground up.

Sorry for the rant. But there are many issues with the technology today and very few people are taking it seriously. It is to the point now where one chooses software based on which is less likely to violate one's private information. But maybe that doesn't matter any more as many large IT corporations have so violated our privacy and sold that data for the almighty dollar and it now makes no difference.

Eran, thank you for your suggestion. I am going to take another attempt to get fsck to cooperate, but not confident in that happening. I did try installing version 17.2 but results exactly the same.

Thank you

Respectfully,

David

**Comment by eranif** [comment]
I can only suggest building wxWidgets with debug symbols and then try and figure out which file caused this. It looks like an invalid INI file format to me.

I think you are on the wrong issue tracker, this should go to the wxWidgets issue tracker (this error is not really related to CodeLite). Moving this into a discussion.
